controllers/postulants.js

```diff
@@ -1,192 +1,92 @@
+    });
+};
 
 const createPostulant = (req, res) => {
-  const params = req.query
-  const newPostulant = {
-    id: params.id,
-    first_name: params.first_name,
-    last_name: params.last_name,
-    email: params.email,
-    phone: params.phone,
-    address: params.address,
-    birthday: params.birthday,
-    password: params.password,
-    primary_studies: {
-      date: {
-        start_date: params.start_date,
-        end_date: params.end_date
-      },
-      school: params.school
-    },
-    secondary_studies: {
-      date: {
-        start_date: params.start_date,
-        end_date: params.end_date
-      },
-      school: params.school
-    },
-    tertiary_studies: {
-      date: {
-        start_date: params.start_date,
-        end_date: params.end_date
-      },
-      institute: params.institute
-    },
-    university_studies: {
-      date: {
-        start_date: params.start_date,
-        end_date: params.end_date
-      },
-      university: params.university
-    },
-    work_experience: [{
-      company: params.company,
-      date: {
-        start_date: params.start_date,
-        end_date: params.end_date
-      },
-      description: params.description
-    }],
-    course: [{
-      institute: params.institute,
-      date: {
-        start_date: params.start_date,
-        end_date: params.end_date
-      },
-      description: params.description
-    }],
-    about: {
-      date: {
-        start_date: params.start_date,
-        end_date: params.end_date
-      },
-      description: params.description
-    },
-    available: params.available
-  }
-
-  Postulants.push(newPostulant)
+  const bodyReq = req.body;
+  const postulant = new Postulants({
+    firstName: bodyReq.firstName,
+    lastName: bodyReq.lastName,
+    email: bodyReq.email,
+    password: bodyReq.password,
+    contactRange: bodyReq.contactRange,
+    address: bodyReq.address,
+    birthday: bodyReq.birthday,
+    available: bodyReq.available,
+    phone: bodyReq.phone,
+    profiles: bodyReq.profiles,
+    studies: bodyReq.studies,
+    workExperience: bodyReq.workExperience,
+  });
 
-  fs.writeFile('./data/postulants.json', JSON.stringify(Postulants), {}, (error) => {
+  postulant.save(((error) => {
     if (error) {
-      res.status(400).send(error)
+      return res.status(400).json({ message: error });
     }
-    return res.status(201).json(newPostulant)
-  })
-}
+    return res.status(201).json(postulant);
+  }));
+};
 
 const deletePostulant = (req, res) => {
-  const params = req.query
-  const filteredPostulants = Postulants.filter(postulant => postulant.id !== params.id)
-
-  if (JSON.stringify(filteredPostulants) === JSON.stringify(Postulants)) {
-    return res.status(404).send('postulant not found')
-  }
-
-  fs.writeFile('./data/postulants.json', JSON.stringify(filteredPostulants), {}, (error) => {
+  Postulants.findByIdAndDelete(req.params.id, (error, pointedPostulant) => {
+    if (!pointedPostulant) {
```

Comment:
Maybe, here the order of errors is important. 

I assume that if an error exists, the new document created `pointedPostulant` will be empty, therefore the code will hardly go into the `if(error)...`. 

It would be safer to invert the order of those if, like

```
Postulants.findByIdAndDelete(req.params.id, (error, pointedPostulant) => {
    if (error) {
      return res.status(400).json({ message: error });
    }
    if (!pointedPostulant) {
      return res.status(404).json({ message: 'Postulant id does not exist' });
    }
    return res.status(204).send();
  });
};
```

Take into account that a non existent Id will not return an error but an empty document (null).

Reply:
fixed!